File: lecture03/phonebook.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

typedef struct
{
    string name;
    string number;
} person;

int main(void)
{
    person people[3];

    people[0].name = "Carter";
    people[0].number = "123";

    people[1].name = "Dave";
    people[1].number = "123";

    people[2].name = "John";
    people[2].number = "567";

    string name = get_string("Name: ");
    for (int i = 0; i < 3; i++)
    {
        if (strcmp(people[i].name, name) == 0)
        {
            printf("Found %s %s\n", name, people[i].number);
            return 0;
        }
    }
    printf("Not found\n");
    return 1;
}

File: week06_python/lecture06/phonebook.py
# names = ["Carter", "David", "John"]

# name = input("Name: ")

# for n in names:
#     if name == n:
#         print("Found")
# else:
#     print("Not found")

# if name in names:
#     print("Found")
# else:
#     print("Not found")

people = [
    {"name": "Carter", "number": "123"},
    {"name": "David", "number": "123"},
    {"name": "John", "number": "456"},
]

people = {
    "Carter": "123",
    "David": "123",
    "John": "456",
}

name = input("Name: ")

# for person in people:
#     if person["name"] == name:
#         print(f"Found {person['number']}")
#         break
# else:
#     print("Not found")

if name in people:
    print(f"Found {people[name]}")
else:
    print("Not found")
